lib/rules/header.js

```diff
@@ -28,46 +28,66 @@ function getLeadingComments(context, node) {
         context.getComments(node).leading;
 }
 
-function genCommentBody(commentType, textArray) {
+function genCommentBody(commentType, textArray, eol) {
     if (commentType === "block") {
         return "/*" + textArray[0] + "*/" + eol;
     } else {
         return "//" + textArray.join(eol + "//") + eol;
     }
 }
 
-function genCommentsRange(context, comments) {
+function genCommentsRange(context, comments, eol) {
     var start = comments[0].range[0];
     var end = comments.slice(-1)[0].range[1];
     if (context.getSourceCode().text[end] === eol) {
-        end++;
+        end += eol.length;
     }
     return [start, end];
 }
 
-function genPrependFixer(commentType, node, headerLines) {
+function genPrependFixer(commentType, node, headerLines, eol) {
     return function(fixer) {
         return fixer.insertTextBefore(
             node,
-            genCommentBody(commentType, headerLines)
+            genCommentBody(commentType, headerLines, eol)
         );
     };
 }
 
-function genReplaceFixer(commentType, context, leadingComments, headerLines) {
+function genReplaceFixer(commentType, context, leadingComments, headerLines, eol) {
     return function(fixer) {
         return fixer.replaceTextRange(
-            genCommentsRange(context, leadingComments),
-            genCommentBody(commentType, headerLines)
+            genCommentsRange(context, leadingComments, eol),
+            genCommentBody(commentType, headerLines, eol)
         );
     };
 }
 
+function findSettings(options) {
+    var lastOption = options.slice(-1)[0];
```

Comment:
Rather than creating a new array, this can be `options[options.length - 1]`

Reply:
@Stuk done :)